pkg/process/metadata/parser/service.go

```diff
@@ -334,3 +336,22 @@ func parseCommandContextJava(args []string) string {
 
 	return "java"
 }
+
+// parseCommandContextDotnet extracts metadata from a dotnet launcher command line
+func parseCommandContextDotnet(args []string) string {
+	for _, a := range args {
+		if strings.HasPrefix(a, "-") {
+			continue
+		}
+		// when running assembly's dll, the cli must be executed without command
+		// https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-run#description
+		if strings.HasSuffix(strings.ToLower(a), ".dll") {
+			_, file := filepath.Split(a)
+			return strings.TrimSuffix(file, filepath.Ext(file))
```

Comment:
we already know the suffix, no need to parse and allocate here

Reply:
I can trim the last 4 chars indeed